frontend/src/components/TE/TaskMixin.ts

```diff
@@ -21,8 +21,7 @@ export default class TaskMixin extends Vue {
   getTaskStyle(task: any) {
     let backgroundColor = "inherit";
     const resolution =
-      task.versions &&
-      _.get(task, `versions[${task.versions.length - 1}].resolution`, false);
+      task.versions && (Object.values(task.versions).pop() as any).resolution;
```

Comment:
I remember we have `LastVersionMixin`. Why not to use it here?
And can you by the way inject the version key to the returned `lastVersion` in order to use it in the `Output` component's download link?

Reply:
`LastVersionMixin` gets the task from the data and this accepts the task as an parameter in the function. Maybe a mixin can be added that they both inherit from that has shared methods? To inject the key should it be added into the object that is returned from `lastVersion` computed property?

Reply:
I think we can add a common ancestor mixin for this.

And yes, the key should be a property of the returned object from `lastVersion`, name it `id`.